Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -266,6 +362,20 @@ void ComponentInfo::failIfScanning() const {
   }
 }
 
+size_t
+ComponentInfo::linearIndex(const std::pair<size_t, size_t> &index) const {
+  // The most common case are beamlines with static components. In that case the
+  // time index is always 0 and we avoid expensive map lookups. Linear indices
+  // are ordered such that the first block contains everything for time index 0
+  // so even in the time dependent case no translation is necessary.
+  if (index.second == 0)
+    return index.first;
+  // Calculate the linear index without a lookup
+  if (m_isSyncScan)
+    return index.first + size() * index.second;
```

Comment:
Is `size()` correct here, since it includes detectors, I think?

Reply:
Good point.